opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -189,6 +231,36 @@ public function index() {
         $this->response->setOutput($this->render());
     }
 
+    protected function handleLayoutSaving() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Check if layout settings are in POST & that it's an array.
+        if (!empty($this->request->post['smaily_for_opencart_module']) && is_array($this->request->post['smaily_for_opencart_module'])) {
+            $smaily_modules_san = [];
+            // Declare layout module settings in POST.
+            $smaily_modules = $this->request->post['smaily_for_opencart_module'];
+            // Loop over each array and validate array keys.
+            foreach ($smaily_modules as $module) {
```

Comment:
You can use different patterns to update elements in array without the need to create a separate array.

Pattern 1 - using PHP references:
```php
$array = array(
  array(
    'foo' => 'bar',
  ),
);
foreach ($array as &$element) {  // Notice the ampersand before variable name.
    $element['foo'] = 'baz';
}
```

Pattern 2 - using array index:
```php
$array = array(
  array(
    'foo' => 'bar',
  ),
);
foreach ($array as $index => $element) {
  $element['foo'] = 'baz';
  $array[$index] = $element;
}
```